central/sensor/service/connection/connection_impl.go

```diff
@@ -597,6 +597,7 @@ func (c *sensorConnection) Run(ctx context.Context, server central.SensorService
 		}
 	}
 
+	metrics.IncrementSensorConnect(c.clusterID, c.sensorHello.GetReconnect())
```

Comment:
What would be the result of communicating with older sensor? Is is guaranteed that `GetReconnect` returns false for old sensors?

Reply:
We might want to add a new sensor capability for this and make use of it (the big list of enums that say what sensor can do currently).

Reply:
> Is is guaranteed that GetReconnect returns false for old sensors?

AFAIK this will return a default value (`false`). 

> We might want to add a new sensor capability for this and make use of it (the big list of enums that say what sensor can do currently).

My idea was to still count the metric even if Sensor is old and reports false. Because it *has* to be false, since these are old sensors, they are not reconnecting for sure. Therefore, I don't see how the capability would help here. At most we could do something like:

```go
if c.HasCapability(MyCompabilityName) {
        metrics.IncrementSensorConnect(c.clusterID, c.sensorHello.GetReconnect())
} else {
        metrics.IncrementSensorConnect(c.clusterID, false)
}
```
Provided it's safe to call `GetReconnect` if the value isn't set, I don't think the capability is needed in this case. 

(Although I do think that we will eventually need a capability entry once we start making changes to which messages are sent on a reconnect as opposed to a fresh start)

Reply:
I was more thinking of 

```go
metrics.IncrementSensorConnect(c.clusterID, c.sensorHello.GetReconnect(), c.HasCapability(MyCompabilityName))
```
(or the FF value instead of capability)

that would give us a full picture.

I consider this change optional as long as the zero-value of that field (bool/enum) will match the old sensors.

Reply:
I wouldn't use the FF value because the FF is currently only used in Sensor. Meaning that it could lead to misconfiguration where the value is set in Sensor but not in Central, resulting in an inconsistent metric point. 

WDYT about making the enum like:
```proto
  enum SensorState {
      UNKNOWN = 0;
      STARTUP = 1;
      RECONNECT = 2;
  }
```
This way, old sensor will always have this value zeroed, and we can mark it is as unknown. New Sensors will either report `STARTUP` or `RECONNECT`. And then we don't need to add the capability, and we have the information if we are connecting to an old or new sensor as well.

Reply:
Sounds great! I like that we have a default value set to something different that the two expected values

Reply:
and I agree on using the FF. I forgot that we are in Central context.